pkg/service/test/match.go

```diff
@@ -88,7 +107,11 @@ func jsonMatch(key string, expected, actual interface{}, noiseMap map[string]boo
 				return false, nil
 			}
 			// remove the noisy key from both expected and actual JSON.
-			if noiseMap[prefix+k] {
+			if regexArr, ok := CheckStringExist(prefix+k, noiseMap); ok {
```

Comment:
This doesn't seem right because we are not sure val is not map. I think recursion of jsonMatch should take care of it.

Reply:
yeah you are right ! My bad I didn't gave a thought.
I have made the changes